include/jemalloc/internal/rtree_types.h

```diff
@@ -25,11 +26,17 @@ typedef struct rtree_s rtree_t;
 #define RTREE_HEIGHT_MAX						\
     ((1U << (LG_SIZEOF_PTR+3)) / RTREE_BITS_PER_LEVEL)
 
+/*
+ * Number of leafkey/leaf pairs to cache.  Each entry supports an entire leaf,
+ * so the cache hit rate is typically high even with a small number of entries.
+ * Two entries is enough to keep the hit rate high even when an arena uses a
+ * combination of dss and mmap.
+ */
+#define RTREE_CTX_NCACHE 2
```

Comment:
I did some experiments with this set to 1, and in one run saw a high miss rate, presumably due to the active memory range straddling two rtree leaf nodes.  The more I think about this, the more I wonder what the appropriate setting is.  As currently configured, each entry supports 1 GiB of virtual memory, which isn't a lot these days...  I'd be tempted to set this to something big (like 128) so that the cache rarely fails even for large applications, but at some point the linear search becomes a problem.  I can come up with justifications for values in the [4..64] range, though that's based on a nebulous notion of how fast linear search (an ```memmove```) is.

Straw poll: What do you think an appropriate setting is, taking into consideration that each entry consumes 16 bytes?